elytron/src/main/java/org/wildfly/extension/elytron/FileSystemRealmDefinition.java

```diff
@@ -139,6 +167,17 @@ private RealmAddHandler() {
             super(SECURITY_REALM_RUNTIME_CAPABILITY, ATTRIBUTES);
         }
 
+        private SecretKey getKey(OperationContext context, String credentialStoreReference, String alias) throws OperationFailedException{
+            ExceptionFunction<OperationContext, CredentialStore, OperationFailedException> credentialStoreApi = context.getCapabilityRuntimeAPI(CREDENTIAL_STORE_API_CAPABILITY, credentialStoreReference, ExceptionFunction.class);
+            CredentialStore credentialStoreResource = credentialStoreApi.apply(context);
+            try {
+                SecretKeyCredential credential = credentialStoreResource.retrieve(alias, SecretKeyCredential.class);
+                return credential.getSecretKey();
```

Comment:
I suppose the returned credential can be `null` in case there is no secret key under the given alias - `java.lang.NullPointerException` is thrown is such case.

For example (no `fs1` secret key in `fs1` credential store)
```
[standalone@embedded /] /subsystem=elytron/filesystem-realm=enc1:add(path=plain1,secret-key=fs1,credential-store=fs1)
{
    "outcome" => "failed",
    "failure-description" => "WFLYCTL0158: Operation handler failed: java.lang.NullPointerException",
    "rolled-back" => true
}
```

Reply:
I misunderstood, changing it to throw a [`credentialDoesNotExist(...)`](https://github.com/wildfly/wildfly-core/blob/main/elytron/src/main/java/org/wildfly/extension/elytron/_private/ElytronSubsystemMessages.java#L377-L378) error if an invalid alias is used.